File: skellig-test-step-processing-jdbc/src/main/java/org/skellig/teststep/processor/jdbc/JdbcRequestExecutorFactory.java
package org.skellig.teststep.processor.jdbc;

import org.skellig.teststep.processor.db.DatabaseRequestExecutor;
import org.skellig.teststep.processor.db.DatabaseRequestExecutorFactory;
import org.skellig.teststep.processor.db.model.DatabaseDetails;
import org.skellig.teststep.processor.jdbc.model.JdbcDetails;

import java.sql.Connection;

class JdbcRequestExecutorFactory extends DatabaseRequestExecutorFactory {

    JdbcRequestExecutorFactory(Connection connection) {
        super(new FindJdbcRequestExecutor(connection), new InsertJdbcRequestExecutor(connection));
    }

    @Override
    public DatabaseRequestExecutor create(DatabaseDetails databaseDetails) {
        return new JdbcRequestExecutor((JdbcDetails) databaseDetails);
    }
}

File: skellig-test-step-processing-jdbc/src/main/kotlin/org/skellig/teststep/processor/jdbc/JdbcRequestExecutorFactory.kt
package org.skellig.teststep.processor.jdbc

import org.skellig.teststep.processor.db.DatabaseRequestExecutorFactory
import java.sql.Connection

internal class JdbcRequestExecutorFactory(connection: Connection?)
    : DatabaseRequestExecutorFactory(
        JdbcSelectRequestExecutor(connection),
        JdbcInsertRequestExecutor(connection),
        JdbcUpdateRequestExecutor(connection,
                JdbcSelectRequestExecutor(connection),
                JdbcInsertRequestExecutor(connection))
)